base/threads_overloads.jl

```diff
@@ -0,0 +1,51 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+"""
+    Threads.foreach(f, channel::Channel,
+                    schedule::Threads.AbstractSchedule=Threads.FairSchedule();
+                    ntasks=Threads.nthreads())
+
+Similar to `foreach(f, channel)`, but iteration over `channel` and calls to
+`f` are split across `ntasks` tasks spawned by `Threads.@spawn`. This function
+will wait for all internally spawned tasks to complete before returning.
+
+If `schedule isa FairSchedule`, `Threads.foreach` will attempt to spawn tasks in a
+manner that enables Julia's scheduler to more freely load-balance work items across
+threads. This approach generally has higher per-item overhead, but may perform
+better than `StaticSchedule` in concurrence with other multithreaded workloads.
+
+If `schedule isa StaticSchedule`, `Threads.foreach` will spawn tasks in a manner
+that incurs lower per-item overhead than `FairSchedule`, but is less amenable
+to load-balancing. This approach thus may be more suitable for fine-grained,
+uniform workloads, but may perform worse than `FairSchedule` in concurrence
+with other multithreaded workloads.
+
+!!! compat "Julia 1.6"
+    This function requires Julia 1.6 or later.
+"""
+function Threads.foreach(f, channel::Channel,
+                         schedule::Threads.AbstractSchedule=Threads.FairSchedule();
```

Comment:
Hmm... How about making `schedule` a keyword argument? I think we might want to have `Threads.foreach(f, arrays...)` at some point? It'd be a bit hard to implement robust dispatch when schedule is at the end of vararg function.

Reply:
Done.